src/test/java/com/commercetools/sync/types/utils/typeactionutils/BuildLocalizedEnumUpdateActionsTest.java

```diff
@@ -0,0 +1,205 @@
+        );
+
+        // remove enum value actions not exists for type resources
+        assertThat(updateActions).containsExactly(
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_D)
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithDifferent_ShouldBuildChangeEnumValueOrderAction() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_CAB
+        );
+
+        assertThat(updateActions).containsExactly(
+            ChangeLocalizedEnumValueOrder.of(FIELD_NAME_1, asList(
+                ENUM_VALUE_C.getKey(),
+                ENUM_VALUE_A.getKey(),
+                ENUM_VALUE_B.getKey()
+            ))
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithRemovedAndDifferentOrder_ShouldBuildChangeOrderAndRemoveActions() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_CB
+        );
+
+        // remove enum value actions not exists for type resources
+        assertThat(updateActions).containsExactly(
+            ChangeLocalizedEnumValueOrder.of(FIELD_NAME_1, asList(
+                ENUM_VALUE_C.getKey(),
+                ENUM_VALUE_B.getKey()
+            ))
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithAddedAndDifferentOrder_ShouldBuildChangeOrderAndAddActions() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_ACBD
+        );
+
+        assertThat(updateActions).containsExactly(
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_D),
+            ChangeLocalizedEnumValueOrder.of(FIELD_NAME_1, asList(
+                ENUM_VALUE_A.getKey(),
+                ENUM_VALUE_C.getKey(),
+                ENUM_VALUE_B.getKey(),
+                ENUM_VALUE_D.getKey()
+            ))
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithAddedEnumValueInBetween_ShouldBuildChangeOrderAndAddActions() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_ADBC
+        );
+
+        assertThat(updateActions).containsExactly(
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_D),
+            ChangeLocalizedEnumValueOrder.of(FIELD_NAME_1, asList(
+                ENUM_VALUE_A.getKey(),
+                ENUM_VALUE_D.getKey(),
+                ENUM_VALUE_B.getKey(),
+                ENUM_VALUE_C.getKey()
+            ))
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithAddedRemovedAndDifOrder_ShouldBuildAllThreeMoveEnumValueActions() {
```

Comment:
ThreeMoveEnum - what do you mean?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/304/commits/f4fc41525cbd101ff25de148431849084815c98a#diff-cf0202967923903ffb0ed23ca81c0c73R188